type_inference_visitor.py

```diff
@@ -0,0 +1,126 @@
+from astroid.transforms import TransformVisitor
+import astroid
+import warnings
+from typing import Tuple, List, Dict
+
+
+class TypeVisitor(TransformVisitor):
+
+    def _visit(self, node):
+        if hasattr(node, '_astroid_fields'):
+            for field in node._astroid_fields:
+                value = getattr(node, field)
+                visited = self._visit_generic(value)
+                setattr(node, field, visited)
+        return self._transform(node)
+
+    def _visit_generic(self, node):
+        if isinstance(node, list):
+            return [self._visit_generic(child) for child in node]
+        elif isinstance(node, tuple):
+            return tuple(self._visit_generic(child) for child in node)
+        else:
+            return self._visit(node)
+
+    # def _visit(self, node):
+    #     # If node is an instance of List/Tuple/Dict, instead of breaking
+    #     # down to smaller pieces of node such as const nodes, keep the
+    #     # bigger piece and directly use an transform function on it.
+    #     if isinstance(node, astroid.List) or isinstance(node, astroid.Tuple)\
+    #             or isinstance(node, astroid.Dict):
+    #         return self._transform(node)
+    #     if hasattr(node, '_astroid_fields'):
+    #         for field in node._astroid_fields:
+    #             value = getattr(node, field)
+    #             visited = self._visit_generic(value)
+    #             setattr(node, field, visited)
+    #     return self._transform(node)
+
+
+def set_const(node):
+    """Populate type constraints for astroid nodes."""
```

Comment:
It would be good to specify the precise node type (`Const`) that this method is for.